tests/unittests/Accelerate/vImageTest.mm

```diff
@@ -585,8 +812,33 @@ static void vImageFillBufferWithRandomDataF(vImage_Buffer* buffer) {
         rowPtr += rgbDest.rowBytes;
     }
 
-    vImageBufferFree(&planarRed);
-    vImageBufferFree(&planarGreen);
-    vImageBufferFree(&planarBlue);
-    vImageBufferFree(&rgbDest);
+    vImageTestBufferFree(&planarRed);
+    vImageTestBufferFree(&planarGreen);
+    vImageTestBufferFree(&planarBlue);
+    vImageTestBufferFree(&rgbDest);
+}
+
+TEST(Accelerate, AlphaUnpremultiply) {
+    SetCACompositor(new NullCompositor);
+
+    char fullPath[_MAX_PATH];
+    GetModuleFileNameA(NULL, fullPath, _MAX_PATH);
+    char* executablePath = strrchr(fullPath, '\\');
+    const char* relativePathToPhoto = "\\Photo2.jpg";
+    strncpy(executablePath, relativePathToPhoto, strlen(relativePathToPhoto) + 1);
+    UIImage* photo = [UIImage imageNamed:[NSString stringWithCString:fullPath]];
+
+    vImage_Buffer unpremultipliedBufferSimd, unpremultipliedBufferNormal;
+    const uint8_t alphaVal = 0x80;
+
+    _vImageSetSimdOptmizationsState(false);
+    vImageTestSetAlphaAndUnpremultiply(photo.CGImage, &unpremultipliedBufferNormal, alphaVal);
+
+    _vImageSetSimdOptmizationsState(true);
+    vImageTestSetAlphaAndUnpremultiply(photo.CGImage, &unpremultipliedBufferSimd, alphaVal);
+
+    vImageTestCompare8888Buffers(&unpremultipliedBufferSimd, &unpremultipliedBufferNormal);
```

Comment:
You're right, a macro would ensure that the line numbers matched up more so than before but it brings with it the potential for naming conflicts when the function exceeds more than a couple of lines. As an alternative to the original implementation and to using a macro, I modified the function to return a bool that says whether or not the buffers match and wrapped the function call in the ASSERT_TRUE macro in the main function so that the line number matched up. The function also prints out a log message indicating the first point at which the buffers don't match.